### PR TITLE
Fixed issue 8 at common repository

### PR DESCRIPTION
**Pull request by XuefengHuang**
Added a function to copy files to subdirectories of the build target dir.

### CODE REVIEW

**Comment by XuefengHuang** [comment]
I forked Priyam's previous PR and make some modifications followed by Vlad's and Albert's suggestion(https://github.com/SeattleTestbed/common/pull/22).
1. Fixed the error in ```config_initialize.txt``` and ```config_build.txt```.
2. Improved readability and delete a unrelated comment in ```build_component.py```.
3. For ```test_addsubdir.py```, adding a function to call ```build_component.py``` directly and limit lines to a maximum of 79 characters. 

This is my test result:

````
Xuefengs-MacBook-Pro:desktop Xuefeng$ git clone https://github.com/XuefengHuang/common
Cloning into 'common'...
remote: Counting objects: 239, done.
remote: Compressing objects: 100% (122/122), done.
remote: Total 239 (delta 121), reused 213 (delta 107)
Receiving objects: 100% (239/239), 61.81 KiB | 0 bytes/s, done.
Resolving deltas: 100% (121/121), done.
Checking connectivity... done.
Xuefengs-MacBook-Pro:desktop Xuefeng$ cd common/scripts
Xuefengs-MacBook-Pro:scripts Xuefeng$ python initialize.py
Checking out repo from https://github.com/SeattleTestbed/common ...
Done!
Checking out repo from https://github.com/SeattleTestbed/utf ...
Done!
Checking out repo from https://github.com/SeattleTestbed/repy_v2 ...
Done!
Checking out repo from https://github.com/SeattleTestbed/seattlelib_v2 ...
Done!
Xuefengs-MacBook-Pro:scripts Xuefeng$ cd ..
Xuefengs-MacBook-Pro:common Xuefeng$ cp build_component.py ./DEPENDENCIES/common/
Xuefengs-MacBook-Pro:common Xuefeng$ cd tests
Xuefengs-MacBook-Pro:tests Xuefeng$ python test_addsubdir.py
Building into /Users/huangxuefeng/Desktop/common/RUNNABLE
Done building!
======================= test result =========================
subdir_all                                            [Pass]
subdir_utf                                            [Pass]
test1                                                 [Pass]
test2                                                 [Pass]
test3                                                 [Pass]
subdir_utf/tests                                      [Pass]
subdir_seattlelib_v2/v2/                              [Pass]
subdir_repy_v2/v2/                                    [Pass]
========================== end ==============================
Xuefengs-MacBook-Pro:tests Xuefeng$ 
````

File: build_component.py [hunk 1]
```diff
@@ -321,16 +313,29 @@ def main():
       # Build instructions for unit tests look like this:
       # "test ../relative/path/to/required/file_or_fileglob"
       if repytest:
-        source_spec = line.split(" ", 1)[1].strip()
+        source_spec = line.split()[1].strip()
+        try:
+          sub_target_dir = line.split()[2].strip()
```
Comment (vladimir-v-diaz) [inline build_component.py]: We should mention this subdirectory feature in the header block. A comment explaining why subdirectories are needed would help.

File: build_component.py [hunk 2]
```diff
@@ -230,7 +223,7 @@ def main():
   # Parse the options provided. 
   helpstring = "python preparetest.py [-t] [-v] [-c] [-r] <target>"
```
Comment (vladimir-v-diaz) [inline build_component.py]: `preparetest.py` is not the correct module name.

File: build_component.py [hunk 3]
```diff
@@ -16,17 +16,16 @@
     to a specific branch, pull from remotes, etc.
     (In a future version of this script, the currently active branch 
     for each repo will be displayed as a visual reminder of this fact.)
-
+    
 <Usage>
   build.py  [-t] [-v] [-c] [-r] <target_directory>
```
Comment (vladimir-v-diaz) [inline build_component.py]: We should explain that `build_component.py` is normally indirectly executed by a script like `build.py`.

File: scripts/config_build.txt [hunk 4]
```diff
@@ -3,6 +3,14 @@
 # there are no non-`test` directives.
 #
 # It is unit-testable however, and here is what we need for this:
-test ./*
-test DEPENDENCIES/utf/*
-test tests/*
+test ./* subdir_all
+test DEPENDENCIES/utf/* subdir_utf
```
Comment (vladimir-v-diaz) [inline scripts/config_build.txt]: Should we move these two test directives to the `# Tests` section below?

File: tests/test_addsubdir.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
+""" 
+test_addsubdir.py --- test if build_component.py can copy files to subdirectories 
+of the build target dir.
+
+The scripts can call build_component.py directly not initialize.py, so you should 
+execute initialize.py before running it.
```
Comment (vladimir-v-diaz) [inline tests/test_addsubdir.py]: I had to re-read this a few times, but I think it's saying that `build.py` should be the only build script that calls `build_component.py`, and the `initialize.py` build script should be manually executed by the user before running `test_addsubdir.py`.  Is that correct?

File: tests/test_addsubdir.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+""" 
+import os 
+import subprocess
+def main():
+	# set component directory
+	component_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+	# change directory to component directory ../../common/
+	os.chdir(component_dir)
+	#subprocess.call(["python", "initialize.py"])
```
Comment (vladimir-v-diaz) [inline tests/test_addsubdir.py]: Should we remove this commented like of code?  Whitespace between comments and code would also help readability.

**Comment by vladimir-v-diaz** [comment]
Is there a reason why this unit test was not written for Seattle's Unit Test Framework?

I think the custom output is fine, but verify with @aaaaalbert.

**Comment by vladimir-v-diaz** [comment]
I think this PR is mergeable once the inline comments above are addressed; let me know when you do so I can perform a final test of the subdirectory functionality.

**Comment by aaaaalbert** [comment]
The "subdir" functionality is available now in 1184e07f3a78b37880a863f72acbcee2c5bf1c30.

@vladimir-v-diaz, I addressed most of your comments in the version I pushed, save for mentioning "subdir" in the docstring. I do point to the build documentation however.

As for the discussion about writing proper unit tests, let's move that to #29.

**Pull request closed without merging**